File: src/data/BovespaLog.php
<?php
namespace src\data;
/**
 * Description of BovespaLog
 *
 * @author Perassoli
 */

class BovespaLog {
    
    public function __construct($log) {
        $this->setLogCode($log);
        $this->setLogDate($log);
        $this->setBDICode($log);
        $this->setTradeCode($log);
        $this->setMarketCode($log);
        $this->setCompanyShortName($log);
        $this->setShareTypeCode($log);
        $this->setMarketTermInDays($log);
        $this->setCurrency($log);
        $this->setOpeningPrice($log);
        $this->setMaxPrice($log);
        $this->setMinPrice($log);
        $this->setMeanPrice($log);
        $this->setLastPrice($log);
        $this->setBestBuyPrice($log);
        $this->setBestSalePrice($log);
        $this->setQttOfTrades($log);
        $this->setQttOfTradedShares($log);
        $this->setVolumeOfTradedShares($log);
        $this->setFactor($log);
    }
    
    public $logCode;
    public function setLogCode($log){
        $this->logCode = substr($log, 0, 2);
    }
    
    public $logDate;
    public function setLogDate($log){
        $this->logDate = substr($log, 2, 8);
    }
    
    public $bdiCode;
    public function setBDICode($log){
        $this->bdiCode = substr($log, 10, 2);
    }
    
    public $tradeCode;
    public function setTradeCode($log){
        $this->tradeCode = substr($log, 12, 12);
    }
    
    public $marketCode;
    public function setMarketCode($log){
        $this->marketCode = substr($log, 24, 3);
    }
    
    public $companyShortName;
    public function setCompanyShortName($log){
        $this->companyShortName = substr($log, 27, 12);
    }
    
    public $shareTypeCode;
    public function setShareTypeCode($log){
        $this->shareTypeCode = substr($log, 39, 10);
    }
    
    public $marketTermInDays;
    public function setMarketTermInDays($log){
        $this->marketTermInDays = substr($log, 49, 3);
    }
    
    public $currency;
    public function setCurrency($log){
        $this->currency = substr($log, 52, 4);
    }
    
    
    public $openingPrice;
    public function setOpeningPrice($log){
        $this->openingPrice = substr($log, 56, 11);
    }
    
    public $maxPrice;
    public function setMaxPrice($log){
        $this->maxPrice = substr($log, 69, 11);
    }
    
    public $minPrice;
    public function setMinPrice($log){
        $this->minPrice = substr($log, 82, 11);
    }
    
    public $meanPrice;
    public function setMeanPrice($log){
        $this->meanPrice = substr($log, 95, 11);
    }
    
    public $lastPrice;
    public function setLastPrice($log){
        $this->lastPrice = substr($log, 108, 11);
    }
    
    public $bestBuyPrice;
    public function setBestBuyPrice($log){
        $this->bestBuyPrice = substr($log, 121, 11);
    }
    
    public $bestSalePrice;
    public function setBestSalePrice($log){
        $this->bestSalePrice = substr($log, 134, 11);
    }
    
    public $qttOfTrades;
    public function setQttOfTrades($log){
        $this->qttOfTrades = substr($log, 147, 5);
    }
    
    
    public $qttOfTradedShares;
    public function setQttOfTradedShares($log){
        $this->qttOfTradedShares = substr($log, 152, 18);
    }
    
    public $volumeOfTradedShares;
    public function setVolumeOfTradedShares($log){
        $this->volumeOfTradedShares = substr($log, 170, 16);
    }
    
    public $factor;
    public function setFactor($log){
        $this->factor = substr($log, 210, 7);
    }
    
}

?>

File: src/data/BovespaLog.java
package data;

public class BovespaLog {

	public BovespaLog(String logTxt) {
		this.setLogCode(logTxt);
		this.setLogDate(logTxt);
		this.setBdiCode(logTxt);
		this.setTradeCode(logTxt);
		this.setMarketCode(logTxt);
		this.setCompanyShortName(logTxt);
		this.setShareTypeCode(logTxt);
		this.setMarketTermInDays(logTxt);
		this.setCurrency(logTxt);
		this.setOpeningPrice(logTxt);
		this.setMaxPrice(logTxt);
		this.setMinPrice(logTxt);
		this.setMeanPrice(logTxt);
		this.setLastPrice(logTxt);
		this.setBestBuyPrice(logTxt);
		this.setBestSalePrice(logTxt);
		this.setQttOfTrades(logTxt);
		this.setQttOfTradedShares(logTxt);
		this.setVolumeOfTradedShares(logTxt);
		this.setFactor(logTxt);
	}

	String logCode;

	void setLogCode(String logTxtTxt) {
		this.logCode = logTxtTxt.substring(0, 2).trim();
	}

	int logDate;

	void setLogDate(String logTxt) {
		this.logDate = Integer.parseInt(logTxt.substring(2, 10).trim());
	}

	String bdiCode;

	void setBdiCode(String logTxt) {
		this.bdiCode = logTxt.substring(10, 12).trim();
	}

	String tradeCode;

	void setTradeCode(String logTxt) {
		this.tradeCode = logTxt.substring(12, 24).trim();
	}

	String marketCode;

	void setMarketCode(String logTxt) {
		this.marketCode = logTxt.substring(24, 27).trim();
	}

	String companyShortName;

	void setCompanyShortName(String logTxt) {
		this.companyShortName = logTxt.substring(27, 39).trim();
	}

	String shareTypeCode;

	void setShareTypeCode(String logTxt) {
		this.shareTypeCode = logTxt.substring(39, 49).trim();
	}

	String marketTermInDays;

	void setMarketTermInDays(String logTxt) {
		this.marketTermInDays = logTxt.substring(49, 52).trim();
	}

	String currency;

	void setCurrency(String logTxt) {
		this.currency = logTxt.substring(52, 56).trim();
	}

	float openingPrice;

	void setOpeningPrice(String logTxt) {
		this.openingPrice = Float.parseFloat(logTxt.substring(56, 67).trim());
	}

	float maxPrice;

	void setMaxPrice(String logTxt) {
		this.maxPrice = Float.parseFloat(logTxt.substring(69, 80).trim());
	}

	float minPrice;

	void setMinPrice(String logTxt) {
		this.minPrice = Float.parseFloat(logTxt.substring(82, 93).trim());
	}

	float meanPrice;

	void setMeanPrice(String logTxt) {
		this.meanPrice = Float.parseFloat(logTxt.substring(95, 106).trim());
	}

	float lastPrice;

	void setLastPrice(String logTxt) {
		this.lastPrice = Float.parseFloat(logTxt.substring(108, 119).trim());
	}

	float bestBuyPrice;

	void setBestBuyPrice(String logTxt) {
		this.bestBuyPrice = Float.parseFloat(logTxt.substring(121, 132).trim());
	}

	String bestSalePrice;

	void setBestSalePrice(String logTxt) {
		this.bestSalePrice = logTxt.substring(134, 145).trim();
	}

	String qttOfTrades;

	void setQttOfTrades(String logTxt) {
		this.qttOfTrades = logTxt.substring(147, 152).trim();
	}

	String qttOfTradedShares;

	void setQttOfTradedShares(String logTxt) {
		this.qttOfTradedShares = logTxt.substring(152, 170).trim();
	}

	String volumeOfTradedShares;

	void setVolumeOfTradedShares(String logTxt) {
		this.volumeOfTradedShares = logTxt.substring(170, 186).trim();
	}

	int factor;

	void setFactor(String logTxt) {
		this.factor = Integer.parseInt(logTxt.substring(210, 217).trim());
	}
}
